src/Servers/Kestrel/Core/src/Internal/Http/ChunkWriter.cs

```diff
@@ -44,6 +44,97 @@ public static int BeginChunkBytes(int dataCount, Span<byte> span)
             return count;
         }
 
+        internal static int GetPrefixBytesForChunk(ref Memory<byte> memory)
+        {
+            var length = memory.Length;
+
+            // If GetMemory returns one of the following values, there is no way to set the prefix/body lengths
+            // such that we either wouldn't have an invalid chunk or would need to copy if the entire memory chunk is used.
+            // For example, if GetMemory returned 21, we would guess that the chunked prefix is 4 bytes initially
+            // and the suffix is 2 bytes, meaning there is 15 bytes remaining to write into. However, 15 bytes only need 3
+            // bytes for the chunked prefix, so we would have to copy once we call advance. Therefore, to avoid this scenario,
+            // we slice the memory by one byte.
+
+            if (length <= 65544)
```

Comment:
Constants for these or not? Don't know which I prefer here.

Reply:
No need for constants IMHO if it's all used in this one method. I wouldn't mind checking in a link to the source that was used to come up with these numbers in case we ever want to tweek the algorithm.

Reply:
I'll include your algorithm just for the shout out.